internal/credentials_test.go

```diff
@@ -17,58 +17,45 @@ package containersuseconnect
 import (
 	"bytes"
 	"log"
-	"strings"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestCredentials(t *testing.T) {
 	cr := &Credentials{}
-
-	if cr.separator() != '=' {
-		t.Fatal("Wrong separator")
-	}
+	require.EqualValues(t, '=', cr.separator())
 
 	prepareLogger()
 	err := cr.afterParseCheck()
+	assert.NotNil(t, err)
 	msg := "Can't find username"
-	if err == nil || err.Error() != msg {
-		t.Fatal("Wrong error")
-	}
+	assert.EqualError(t, err, msg)
 	shouldHaveLogged(t, msg)
 
 	cr.setValues("username", "suse")
 	prepareLogger()
-	msg = "Can't find password"
 	err = cr.afterParseCheck()
-	if err == nil || err.Error() != msg {
-		t.Fatal("Wrong error")
-	}
+	assert.NotNil(t, err)
+	msg = "Can't find password"
+	assert.EqualError(t, err, msg)
 	shouldHaveLogged(t, msg)
 
 	cr.setValues("password", "1234")
 	err = cr.afterParseCheck()
-	if err != nil {
-		t.Fatal("There should not be an error")
-	}
+	assert.Nil(t, err)
 
 	locs := cr.locations()
-	if locs[0] != "/etc/zypp/credentials.d/SCCcredentials" {
-		t.Fatal("Wrong location")
-	}
-	if locs[1] != "/run/secrets/SCCcredentials" {
-		t.Fatal("Wrong location")
-	}
-	if locs[2] != "/run/secrets/credentials.d/SCCcredentials" {
-		t.Fatal("Wrong location")
-	}
+	assert.Contains(t, locs, "/etc/zypp/credentials.d/SCCcredentials")
+	assert.Contains(t, locs, "/run/secrets/SCCcredentials")
+	assert.Contains(t, locs, "/run/secrets/credentials.d/SCCcredentials")
```

Comment:
this isn't the same. previously it was checking for a specific ordering of the locations, now you are accepting it at any order. given that lookup order matters, why are we not checking on locs[0] with assertequal instead?

Reply:
If the order is important, then the test should check it accordingly. I though that the order was not that critical.

Reply:
* the previous test was testing ordering
* the ordering is important to validate that configs are read in the right order (so that first or last one wins, whatever is implemented)